nixos/modules/services/networking/headscale.nix

```diff
@@ -424,33 +487,45 @@ in {
     (mkRenamedOptionModule ["services" "headscale" "tls" "letsencrypt" "httpListen"] ["services" "headscale" "settings" "tls_letsencrypt_listen"])
     (mkRenamedOptionModule ["services" "headscale" "tls" "certFile"] ["services" "headscale" "settings" "tls_cert_path"])
     (mkRenamedOptionModule ["services" "headscale" "tls" "keyFile"] ["services" "headscale" "settings" "tls_key_path"])
-    (mkRenamedOptionModule ["services" "headscale" "aclPolicyFile"] ["services" "headscale" "settings" "acl_policy_path"])
+    (mkRenamedOptionModule ["services" "headscale" "aclPolicyFile"] ["services" "headscale" "settings" "policy" "path"])
+
+    # TODO(kradalby): figure out why these are not working?
+    # Since 0.23.0
+    # (mkRenamedOptionModule ["services" "headscale" "settings" "db_type"] ["services" "headscale" "settings" "database" "type"])
```

Comment:
I could not make this work, I am not sure why, got messages like:

```
error: evaluation aborted with the following error message: 'cannot find attribute `services.headscale.settings.dns_config.base_domain`
```

But I didnt expect it to have to exist?

Reply:
I don't think you can rewrite settings through multiple renamedOptionModule. Just delete the old ones and only add the ones required from 0.22.0 to 0.23.0

Reply:
ah, it looks like you cannot use mkRenamedOptionModule on dynamic options like settings...

Reply:
You should be able if the target option is defined inside the submodule.

Reply:
I am getting errors whenever I use the function on something that is not defined, are you able to reproduce it?

Reply:
Nope. I just tried it locally and nothing like that happened. I am just going to push the uncomment on top and we see what ofborg and the others say.

Reply:
Now I understand what is going on. We cannot have a rename for an option which exists in the path of the new options.

eg: settings.dns_config.nameservers -> settings.dns_config.nameservers.global doesn't work because nameservers exists before and after.

Reply:
Found it https://github.com/NixOS/nixpkgs/blob/cd9c62afca390a4b9390e526a495736d183ca6e1/nixos/modules/programs/tsm-client.nix#L102-L111 🎉